planning/behavior_path_planner/src/scene_module/avoidance/avoidance_utils.cpp

```diff
@@ -60,6 +64,147 @@ geometry_msgs::msg::Polygon toMsg(const tier4_autoware_utils::Polygon2d & polygo
   return ret;
 }
 
+bool validCheckDecelPlan(
+  const double v_end, const double a_end, const double v_target, const double a_target,
+  const double v_margin, const double a_margin)
+{
+  const double v_min = v_target - std::abs(v_margin);
+  const double v_max = v_target + std::abs(v_margin);
+  const double a_min = a_target - std::abs(a_margin);
+  const double a_max = a_target + std::abs(a_margin);
+
+  if (v_end < v_min || v_max < v_end) {
+    RCLCPP_DEBUG_STREAM(
+      rclcpp::get_logger("validCheckDecelPlan"),
+      "[validCheckDecelPlan] valid check error! v_target = " << v_target << ", v_end = " << v_end);
+    return false;
+  }
+  if (a_end < a_min || a_max < a_end) {
+    RCLCPP_DEBUG_STREAM(
+      rclcpp::get_logger("validCheckDecelPlan"),
+      "[validCheckDecelPlan] valid check error! a_target = " << a_target << ", a_end = " << a_end);
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * @brief calculate distance until velocity is reached target velocity (TYPE1)
+ * @param (v0) current velocity [m/s]
+ * @param (vt) target velocity [m/s]
+ * @param (a0) current acceleration [m/ss]
+ * @param (am) minimum deceleration [m/ss]
+ * @param (ja) maximum jerk [m/sss]
+ * @param (jd) minimum jerk [m/sss]
+ * @param (t_min) duration of constant deceleration [s]
+ * @return moving distance until velocity is reached vt [m]
+ * @detail TODO(Satoshi Ota)
+ */
+boost::optional<double> calcDecelDistPlanType1(
```

Comment:
FYI: Because the difference of Type1,2,3 is just an integration time, it can be more simplified. The following code should (maybe) works for all type situations.

```cpp
  double x0 = 0.0;

  const auto [t_dec_jerk, t_zero_jerk, t_acc_jerk] = []() {
    const auto v_end = 0.0;
    const auto t_zero_jerk = (v_end - v0 + 0.5 * (a0*a0 - amin*amin) / jd + (0.5 * amin*amin / ja)) / amin;

    if (t_zero_jerk > 0) {
      // jerk_plan_type = decel -> zero -> acc
      const auto t_dec_jerk = (amin - a0) / jd;
      const auto t_acc_jerk = -amin / ja;
      return std::make_tuple(t_dec_jerk, t_zero_jerk, t_acc_jerk);
    } else {
      const double min_acc_actual = -sqrt(2.0 * (v_end - v0 + (0.5 * a0*a0 / jd)) * ((jd * ja) / (ja - jd)));
      if (a0 > min_acc_actual) {
        // jerk_plan_type = decel -> acc
        const auto t_dec_jerk = (min_acc_actual - a0) / jd;
        const auto t_acc_jerk = -min_acc_actual / ja;
      return std::make_tuple(t_dec_jerk, t_zero_jerk, t_acc_jerk);
      } else {
        // jerk_plan_type = acc
        const auto t_dec_jerk = 0.0;
        const auto t_acc_jerk = -a0 / ja;
        return std::make_tuple(t_dec_jerk, t_zero_jerk, t_acc_jerk);
      }
    }
  }();

  const auto [x1, v1, a1] = update(x0, v0, a0, jd, t_dec_jerk);
  const auto [x2, v2, a2] = update(x1, v1, a1, 0.0, t_zero_jerk);
  const auto [x3, v3, a3] = update(x2, v2, a2, ja, t_acc_jerk);
```